### ISSUE TITLE
Create multiple certificates

### ISSUE THREAD

**Issue by alex88** [janeczku/rancher-letsencrypt#16]
Hi there, not sure if i can be done or if it's worth it, but do you have any plans to create multiple certificates instead of having one with multiple hostnames?
I think it could be useful for big projects where a company would like to not expose other services domains in the certificate.

I imagine having the same comma-separated list of domains but instead of having everything in a certificate, use the certificate field as a prefix for the new certificates it's going to create

**Comment by zambon** [comment]
@alex88 Have you considered deploying a Let's Encrypt stack for the first certificate, and then add new services (or clone and modify) for the following certificates?

It may be a lot of work, but it'd be easy to manage in Rancher.

**Comment by alex88** [comment]
@zambon that's what I'm doing right now, I'm just wondering what will happen when a new version is released :D we've now 20 instances running

**Comment by janeczku** [comment]
Sound like a reasonable enhancement.
What about an option to make it create one certificate per given domain instead of a SAN certificate. And the resulting single-domain certs would be named using the value of the certificate name field suffixed with the domain. Thoughts @alex88 @zambon?

**Comment by alex88** [comment]
How about using a single docker containers with the domains in one single variable, with each certificate separated by semicolon and each domain by a comma? Too complex for the user?
Also creating n certificates in rancher with a prefix and using the first certificate name as the rancher cert name.

e.g.:

```
site1.com,www.site1.com;site2.com,www.site2.com,site3.com

certs in rancher: <prefix>-site1.com, <prefix>-site2.com
domains in <prefix>-site1.com: site1.com,www.site1.com
domains in <prefix>-site2.com: site2.com,www.site2.com,site3.com
```

**Comment by zambon** [comment]
@janeczku That sounds good! Having the _option_ to choose is essential though, as larger deployments may cause the service to run into rate limits.

@alex88 I like your idea too. It may be simpler if the catalog used a `multiline` textbox, then you could have a line-based separation, instead of semicolon-base. E.g.:
````
site1.com, www.site1.com
site2.com, www.site2.com, site3.com
```

BTW @alex88, I don't mean to sidetrack into another conversation, but how to you handle having many certificates? Do you run one (set of) load balancer(s) per stack? _(I'm currently using a "global" load balancer in my deployments, which means I need to have a single certificate)_

**Comment by alex88** [comment]
@zambon I have a single load balancer running on every instance which has a main certificate and multiple others, SNI support is good enough for us, so it sends the right certificate for the right domain

**Comment by janeczku** [comment]
@alex88 Yeah that was the other option. Using a multiline text field as suggested by @zambon  seems like the best way to go about this.

@zambon Rate-limits will be an issue, both on the LE CA side and for some of the DNS providers.

**Comment by alex88** [comment]
@janeczku are there multiline textboxes when creating it? The problem comes later when you've to upgrade the env variables and the input is not multiline

**Comment by janeczku** [comment]
@alex88 good thinking. Is this something you are regulary doing? I mean upgrading the env vars to change the certificates?

**Comment by alex88** [comment]
Well I did it every 4-5 days now that we just started using letsencrypt and so we've migrated many certificates.
Otherwise nope, it's not something common. Btw, I've just tested and it supports multiline (at least the env variable field) so I agree, multiline is easier to read and to edit

![screen shot 2016-07-13 at 4 31 12 pm](https://cloud.githubusercontent.com/assets/436888/16823217/4b033d72-4917-11e6-95da-e81618c844a6.png)

**Comment by alex88** [comment]
Catalog templates also support multiline inputs as textarea, so it should work

![screen shot 2016-07-13 at 4 33 28 pm](https://cloud.githubusercontent.com/assets/436888/16823257/974ff53a-4917-11e6-900b-3d4306d60576.png)

**Comment by ffittschen** [comment]
@janeczku I would agree with @zambon, that it is necessary to be able to choose whether to have certificates with SANs or single-domain certificates. But the multiline idea would solve that quite well:
* for certificates with SANs one would have one certificate per line, CN at the front and SANs following
* for single-domain certificates one would write every domain in a new line

Regarding the rate-limits: Since December 2016 there are [additional rate limits](https://letsencrypt.org/docs/rate-limits/) to the *cert per week*-limits. The most important one would be
> The “new-reg”, “new-authz” and “new-cert” endpoints have an **Overall Request Per Second** rate limit of 20 requests per second. All other endpoints have a rate limit of 2000 requests per second.

**Comment by gizmotronic** [comment]
I'd like to be able to mix and match single-domain certificates and certificates with SANs. For now I have three separate containers, each with an identical configuration (included shared storage volume) except for the certificate name, domain names, and renewal time.

**Comment by iBobik** [comment]
How about to get domains list from the load balancer? #67

**Comment by maxfriedmann** [comment]
Any updates on this? I really like  the suggestion from @alex88, its how we used to work with the dockercloud & JrCs/docker-letsencrypt-nginx-proxy-companion, but #67 would be the greatest solution!

**Comment by iBobik** [comment]
As an alternative I recommend this: https://github.com/adi90x/rancher-active-proxy

Jan Pobořil
https://honza.poboril.cz/

4. 3. 2018 v 6:56, Maximilian Friedmann <notifications@github.com>:

**Comment by karael** [comment]
> Any updates on this? I really like the suggestion from @alex88, its how we used to work with the dockercloud & JrCs/docker-letsencrypt-nginx-proxy-companion, but #67 would be the greatest solution!

I agree
